02-peer-protocol.md

```diff
@@ -1713,6 +1784,181 @@ satoshis, which is possible if `dust_limit_satoshis` is below 546 satoshis).
+      - MUST use `closer_output_only`.
+    - Otherwise, if it considers the local output amount uneconomical AND its `closee_scriptpubkey` is not `OP_RETURN`:
+      - MUST use `closer_output_only`.
+    - Otherwise, if `closer_and_closee_outputs` is present:
+      - MUST use `closer_and_closee_outputs`.
+    - Otherwise:
+      - MUST use `closee_output_only`.
+  - If the selected signature field does not exist:
+    - MUST either send a `warning` and close the connection, or send an `error` and fail the channel.
+  - If the signature field is not valid for the corresponding closing transaction specified in [BOLT #3](03-transactions.md#closing-transaction):
+    - MUST either send a `warning` and close the connection, or send an `error` and fail the channel.
+  - If the signature field is non-compliant with LOW-S-standard rule<sup>[LOWS](https://github.com/bitcoin/bitcoin/pull/6769)</sup>:
+    - MUST either send a `warning` and close the connection, or send an `error` and fail the channel.
+  - MUST sign and broadcast the corresponding closing transaction.
+  - MUST send `closing_sig` with a single valid signature in the same TLV field as the `closing_complete`.
+  - MUST use `closer_scriptpubkey` for its own future `closing_complete` messages.
+
+The receiver of `closing_sig`:
+  - If `closer_scriptpubkey`, `closee_scriptpubkey`, `fee_satoshis` or `locktime` don't match what was sent in `closing_complete`:
+    - MUST either send a `warning` and close the connection, or send an `error` and fail the channel.
+  - If `tlvs` does not contain exactly one signature:
+    - MUST either send a `warning` and close the connection, or send an `error` and fail the channel.
+  - If `tlvs` does not contain one of the TLV fields sent in `closing_complete`:
+    - MUST either send a `warning` and close the connection, or send an `error` and fail the channel.
+  - If the signature field is not valid for the corresponding closing transaction specified in [BOLT #3](03-transactions.md#closing-transaction):
+    - MUST either send a `warning` and close the connection, or send an `error` and fail the channel.
+  - If the signature field is non-compliant with LOW-S-standard rule<sup>[LOWS](https://github.com/bitcoin/bitcoin/pull/6769)</sup>:
+    - MUST either send a `warning` and close the connection, or send an `error` and fail the channel.
+  - otherwise:
+    - MUST broadcast the corresponding closing transaction.
+  - MAY send another `closing_complete` (e.g. with a different `fee_satoshis` or `closer_scriptpubkey`).
+
+### Rationale
+
+The close protocol is designed to avoid any failure scenarios caused by fee disagreement,
+since each side offers to pay its own desired fee.
+
+If one side has less funds than the other, it may choose to omit its own output, and in this case
+dust MUST be omitted, to ensure that the resulting transaction can be broadcast.
+
+The corner case where fees are so high that both outputs are dust is addressed in two ways: paying
```

Comment:
Reading this again, doesn't this allow one party to just entirely remove the other party's output on the closing transaction? If Bob can unilaterally decide that both outputs are dust, then he can burn my output by sending an `OP_RETURN` script.